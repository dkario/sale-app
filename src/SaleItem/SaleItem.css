.sale-item {
    margin: 50px;
}

.image {
    width: 120px;
    height: 120px;
}

.base-price {
    text-decoration: line-through;
}

.sale-price {
    font-weight: bolder;
    color: red;
}
