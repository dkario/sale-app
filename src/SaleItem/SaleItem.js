import React, { Component } from 'react';
import './SaleItem.css';

class SaleItem extends Component {
    // Step 0: Take a look at the successfully compiled app in your browser,
    // and then at the logic used inside `render()` below.
    // Once you get the gist of it, follow the instructions in `render()`.

    // Step 1: implement this business logic.
    calculateSalePrice(basePrice, salePercentage) {
        // TODO
    }

    // Step 2: implement this view logic.
    formatPriceAsDollar(price) {
        // TODO
    }

    // Step 3: Implement this by calling helper method(s) above with `this.props`.
    getBasePrice() {
        // TODO
    }

    // Step 4: Implement this by calling helper method(s) above with `this.props`.
    getSalePrice() {
        // TODO
    }

    render() {
        return (
          <section className="sale-item">
            <img src="plant.jpg" alt="plant" className="image" />
            <p>
                Buy this plant for only
                { ' ' }
                <span className="base-price">
                    {/* { this.getBasePrice() } */}
                    {/* Uncomment above and extract below into refactored class methods */}
                    { '$' + parseFloat(Math.round(this.props.basePrice * 100) / 100).toFixed(2) }
                </span>
                { ' ' }
                <span className="sale-price">
                    {/* { this.getSalePrice() } */}
                    {/* Uncomment above and extract below into refactored class methods */}
                    { '$' + parseFloat(Math.round(this.props.basePrice * (1 - this.props.salePercentage) * 100) / 100).toFixed(2) }
                </span>
                !
            </p>
            </section>
        );
    }
}

export default SaleItem;
