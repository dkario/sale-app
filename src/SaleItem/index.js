import SaleItem from './SaleItem';

export default SaleItem;
