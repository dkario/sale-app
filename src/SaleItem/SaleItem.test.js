import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import SaleItem from './SaleItem';

describe('SaleItem component', () => {
    it('matches snapshot', () => {
        const wrapper = shallow(<SaleItem basePrice="60" salePercentage="0.25" />);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders $45 sale price for a $60 item on a 25% sale', () => {
        const saleItem = shallow(<SaleItem basePrice="60" salePercentage="0.25" />);

        expect(saleItem.text()).toEqual('Buy this plant for only $60.00 $45.00!');
    });

    describe('calculateSalePrice', () => {
        it('should calculate and return the sale price by taking the `salePercetange` off of the `basePrice`', () => {
            const actual = SaleItem.prototype.calculateSalePrice(60, 0.25);
            const expected = 45;

            expect(actual).toEqual(expected);
        });
    });

    describe('formatPriceAsDollar', () => {
        it('should return the integer argument formatted as a dollar amount', () => {
            const actual = SaleItem.prototype.formatPriceAsDollar(60);
            const expected = '$60.00';

            expect(actual).toEqual(expected);
        });
    });
});
