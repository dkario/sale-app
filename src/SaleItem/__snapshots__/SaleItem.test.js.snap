// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaleItem component matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SaleItem
    basePrice="60"
    salePercentage="0.25"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="plant"
          className="image"
          src="plant.jpg"
/>,
        <p>
          Buy this plant for only
           
          <span
                    className="base-price"
          >
                    $60.00
          </span>
           
          <span
                    className="sale-price"
          >
                    $45.00
          </span>
          !
</p>,
      ],
      "className": "sale-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "alt": "plant",
          "className": "image",
          "src": "plant.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Buy this plant for only",
            " ",
            <span
              className="base-price"
>
              $60.00
</span>,
            " ",
            <span
              className="sale-price"
>
              $45.00
</span>,
            "!",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Buy this plant for only",
          " ",
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "$60.00",
              "className": "base-price",
            },
            "ref": null,
            "rendered": "$60.00",
            "type": "span",
          },
          " ",
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "$45.00",
              "className": "sale-price",
            },
            "ref": null,
            "rendered": "$45.00",
            "type": "span",
          },
          "!",
        ],
        "type": "p",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="plant"
            className="image"
            src="plant.jpg"
/>,
          <p>
            Buy this plant for only
             
            <span
                        className="base-price"
            >
                        $60.00
            </span>
             
            <span
                        className="sale-price"
            >
                        $45.00
            </span>
            !
</p>,
        ],
        "className": "sale-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "alt": "plant",
            "className": "image",
            "src": "plant.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Buy this plant for only",
              " ",
              <span
                className="base-price"
>
                $60.00
</span>,
              " ",
              <span
                className="sale-price"
>
                $45.00
</span>,
              "!",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Buy this plant for only",
            " ",
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "$60.00",
                "className": "base-price",
              },
              "ref": null,
              "rendered": "$60.00",
              "type": "span",
            },
            " ",
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "$45.00",
                "className": "sale-price",
              },
              "ref": null,
              "rendered": "$45.00",
              "type": "span",
            },
            "!",
          ],
          "type": "p",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
